src/withAxiom.ts

```diff
@@ -124,13 +133,44 @@ function withAxiomNextApiHandler(handler: NextApiHandler): NextApiHandler {
   };
 }
 
+type ParsedUrlQuery = GetServerSidePropsContext['query'];
+export type AxiomContext<
+  Q extends ParsedUrlQuery = ParsedUrlQuery,
+  D extends PreviewData = PreviewData
+> = GetServerSidePropsContext<Q, D> & { log: Logger };
+export type AxiomGetServerSideProps<
+  P extends { [key: string]: any } = { [key: string]: any },
+  Q extends ParsedUrlQuery = ParsedUrlQuery,
+  D extends PreviewData = PreviewData
+> = (context: AxiomContext<Q, D>) => Promise<GetServerSidePropsResult<P>>;
+
+export function withAxiomNextServerSidePropsHandler(handler: AxiomGetServerSideProps): GetServerSideProps {
+  return async (context) => {
+    const report: RequestReport = config.generateRequestMeta(context.req);
```

Comment:
I think the report will not be correct, because the context won't have url and method, they will be under `ctx.req`. 

maybe we will need a separate function to generate report for the getServerSideProps.